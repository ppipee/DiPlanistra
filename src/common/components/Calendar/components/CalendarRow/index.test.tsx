import React from 'react'

import { render } from '@testing-library/react'
import 'jest-styled-components'

import * as CalendarRowType from '.'

describe('Calendar > CalendarRow', () => {
	const CalendarCellSpy = jest.fn()

	jest.doMock('common/components/Calendar/components/CalendarCell', () => (props: Record<string, any>) => {
		CalendarCellSpy(props)

		return <div {...props}>{props.children}</div>
	})

	const { default: CalendarRow } = require('.') as typeof CalendarRowType

	afterEach(() => {
		jest.resetAllMocks()
	})

	it('should render CalendarRow width Dates', () => {
		const { container } = render(
			<CalendarRow
				row={[
					new Date(2020, 0, -1),
					new Date(2020, 0, 0),
					new Date(2020, 0, 1),
					new Date(2020, 0, 2),
					new Date(2020, 0, 3),
					new Date(2020, 0, 4),
					new Date(2020, 0, 5),
				]}
			/>,
		)

		expect(container.firstChild).toMatchSnapshot()
	})
})
