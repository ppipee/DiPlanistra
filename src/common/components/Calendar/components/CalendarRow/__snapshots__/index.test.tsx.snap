// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar > CalendarRow should render CalendarRow width Dates 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="c0"
>
  <div
    date="Mon Dec 30 2019 00:00:00 GMT+0700 (Indochina Time)"
    index="0"
  />
  <div
    date="Tue Dec 31 2019 00:00:00 GMT+0700 (Indochina Time)"
    index="1"
  />
  <div
    date="Wed Jan 01 2020 00:00:00 GMT+0700 (Indochina Time)"
    index="2"
  />
  <div
    date="Thu Jan 02 2020 00:00:00 GMT+0700 (Indochina Time)"
    index="3"
  />
  <div
    date="Fri Jan 03 2020 00:00:00 GMT+0700 (Indochina Time)"
    index="4"
  />
  <div
    date="Sat Jan 04 2020 00:00:00 GMT+0700 (Indochina Time)"
    index="5"
  />
  <div
    date="Sun Jan 05 2020 00:00:00 GMT+0700 (Indochina Time)"
    index="6"
  />
</div>
`;
